cpp/ql/src/Likely Bugs/Underspecified Functions/MistypedFunctionArguments.ql

```diff
@@ -72,12 +83,10 @@ where
   p = f.getAParameter() and
   not f.isVarargs() and
   p.getIndex() < fc.getNumberOfArguments() and
-  // There must be a zero-parameter declaration
-  exists(FunctionDeclarationEntry fde | fde = f.getADeclarationEntry() |
-    fde.getNumberOfParameters() = 0
-  ) and
+  hasZeroParamDecl(f) and
   // Parameter p and its corresponding call argument must have mismatched types
   not argMayBeUsed(fc.getArgument(p.getIndex()), p)
-select fc, "Calling $@: argument $@ of type $@ is incompatible with parameter $@.", f, f.toString(),
-  fc.getArgument(p.getIndex()) as arg, arg.toString(), arg.getFullyConverted().getType() as type,
-  type.toString(), p, p.getTypedName()
+select fc, "Calling $@: argument $@ of type $@ is incompatible with parameter $@", f, f.toString(),
```

Comment:
We prefer a full stop after alert messages. See https://github.com/Semmle/ql/blob/master/docs/query-metadata-style-guide.md#alert--messages